src/panoptes/pocs/focuser/birger.py

```diff
@@ -249,13 +249,15 @@ def _send_command(self, command, response_length=None):
         # Clear the input buffer in case there's anything left over in there.
         self._serial_port.reset_input_buffer()
 
-        # Send command
-        self._serial_io.write(command + '\r')
-
         # In verbose mode adaptor will first echo the command
-        echo = self._serial_io.readline().rstrip()
-        assert echo == command, self.logger.warning("echo != command: {} != {}".format(
-            echo, command))
+        echo = [self._serial_io.readline().rstrip()]
+        while echo[0] != command:
+            # Send command
+            self._serial_io.write(command + '\r')
+            echo[0] = self._serial_io.readline().rstrip()
+
+        assert echo[0] == command, self.logger.warning("echo != command: {} != {}".format(
+            echo[0], command))
```

Comment:
They way you currently have the loop working this is totally redundant code, as there is no way get out of the loop until `echo[0] == command` is `True`.